projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts

```diff
@@ -550,20 +546,18 @@ export class IgxFilteringService implements OnDestroy {
         const newExpressionsTree: FilteringExpressionsTree = createNewTree ?
             new FilteringExpressionsTree(filteringState.operator, filteringState.fieldName) : filteringState as FilteringExpressionsTree;
 
-        if (oldExpressionsTreeIndex === -1) {
-            // no expressions tree found for this field
-            if (expressionsTree) {
-                if (insertAtIndex > -1) {
-                    newExpressionsTree.filteringOperands.splice(insertAtIndex, 0, expressionsTree);
-                } else {
-                    newExpressionsTree.filteringOperands.push(expressionsTree);
-                }
-            } else if (condition) {
-                // create expressions tree for this field and add the new expression to it
-                const newExprTree: FilteringExpressionsTree = new FilteringExpressionsTree(filteringState.operator, fieldName);
-                newExprTree.filteringOperands.push(newExpression);
-                newExpressionsTree.filteringOperands.push(newExprTree);
+        // no expressions tree found for this field
+        if (expressionsTree) {
+            if (insertAtIndex > -1) {
+                newExpressionsTree.filteringOperands.splice(insertAtIndex, 1, expressionsTree);
```

Comment:
Could we simplify it like this?
```suggestion
                newExpressionsTree.filteringOperands[insertAtIndex] = expressionsTree;
```